tokio/src/runtime/mod.rs

```diff
@@ -408,10 +409,10 @@ impl Runtime {
     /// complete, and yielding its resolved result. Any tasks or timers which
     /// the future spawns internally will be executed on the runtime.
     ///
-    /// `&mut` is required as calling `block_on` **may** result in advancing the
-    /// state of the runtime. The details depend on how the runtime is
-    /// configured. [`runtime::Handle::block_on`][handle] provides a version
-    /// that takes `&self`.
+    /// When this runtime is configured with `core_threads = 0` only the first call
+    /// to `block_on` will run the io/timer driver. All other calls _before_ the first
+    /// `block_on` completes will just hook into that driver. This means the driver
+    /// may be passed around from thread to thread manually by the user.
```

Comment:
```suggestion
    /// `block_on` completes will just hook into the driver running on the thread
    /// that first called `block_on`. This means that the driver may be passed
    /// from thread to thread by the user between calls to `block_on`.
```
again, assuming this is what you meant here.

Reply:
also, it might be nice to have examples explaining this behavior, it's a little hard to explain with just text?

Reply:
+1 to Eliza's suggestion, but I'd remove mentions of "just".

Reply:
Yeah, will write examples in other PRs 👍 and for the site.